qiskit_experiments/quantum_volume/qv_analysis.py

```diff
@@ -0,0 +1,276 @@
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Quantum Volume analysis class.
+"""
+
+import math
+import warnings
+from typing import Optional
+import numpy as np
+
+from qiskit_experiments.base_analysis import BaseAnalysis
+from qiskit_experiments.base_analysis import AnalysisResult
+from qiskit_experiments.analysis.plotting import (
+    HAS_MATPLOTLIB,
+    plot_scatter,
+    plot_errorbar,
+)
+
+
+class QVAnalysis(BaseAnalysis):
+    """RB Analysis class."""
+
+    # pylint: disable = arguments-differ
+    def _run_analysis(self, experiment_data, plot: bool = True, ax: Optional["AxesSubplot"] = None):
+        """Run analysis on circuit data.
+        Args:
+            experiment_data (ExperimentData): the experiment data to analyze.
+            plot: If True generate a plot of fitted data.
+            ax: Optional, matplotlib axis to add plot to.
+        Returns:
+            tuple: A pair ``(analysis_result, figures)`` where
+                   ``analysis_results`` may be a single or list of
+                   AnalysisResult objects, and ``figures`` may be
+                   None, a single figure, or a list of figures.
+        """
+        depth = experiment_data.experiment.num_qubits
+        num_trials = experiment_data.experiment.trials
+        data = experiment_data.data()
+
+        heavy_outputs = np.zeros(num_trials, dtype=list)
+        heavy_output_prob_exp = np.zeros(num_trials, dtype=list)
+
+        # analyse ideal data to calculate all heavy outputs
+        # must calculate first the ideal data, because the non-ideal calculation uses it
+        for ideal_data in data:
+            if not ideal_data["metadata"].get("is_simulation", None):
+                continue
+            trial = ideal_data["metadata"]["trial"]
+            trial_index = trial - 1  # trials starts from 1, so as index use trials - 1
+
+            heavy_outputs[trial_index] = self._calc_ideal_heavy_output(ideal_data)
+
+        # analyse non-ideal data
+        for non_ideal_data in data:
+            if non_ideal_data["metadata"].get("is_simulation", None):
+                continue
+            trial = non_ideal_data["metadata"]["trial"]
+            trial_index = trial - 1  # trials starts from 1, so as index use trials - 1
+
+            heavy_output_prob_exp[trial_index] = self._calc_exp_heavy_output_probability(
+                non_ideal_data, heavy_outputs[trial_index]
+            )
+
+        analysis_result = AnalysisResult(
+            self._calc_quantum_volume(heavy_output_prob_exp, depth, num_trials)
+        )
+
+        if plot and HAS_MATPLOTLIB:
```

Comment:
Does this plot can also be useful for the QV experiment?
Qiskit/qiskit-ignis#489
Can you have more than one plot?

Reply:
this plot might be useful, and I can have more than one plot.
the problem is such a plot is for a specific trial. in high QV there are typically hundreds of trials, so creating a plot for each is too much.
maybe we can think of a way to add a method to the experiment to create such plot after analysis given specific trial

Reply:
I would be happy to get some feedback of the experimentalists about the plots they would like to see and the user experience